server/src/main/java/org/elasticsearch/index/mapper/RoutingFieldMapper.java

```diff
@@ -102,6 +103,11 @@ public boolean required() {
     @Override
     public void preParse(DocumentParserContext context) {
         String routing = context.sourceToParse().routing();
+        if (context.indexSettings().mode() == IndexMode.TIME_SERIES) {
+            // TODO when we stop storing the tsid in the routing fail any request with routing in time series mode
```

Comment:
I have been thinking about that. Allowing routing in the time_series mode might actually make sense. Yes, it is allowing users to shoot themselves in the foot. but routing is an advanced feature that already enables screwing  things up on "standard" indices, but it also enables advanced users to build solutions they wouldn't be able to build otherwise.

Having only few elements of TSID in the routing key doesn't break anything, but it can speed things up and help in case of multi-tenant solutions.

Reply:
I suppose so. I can revive support for routing once we have a field on `IndexRequest` specifically for the tsid. That was something I was hoping to do in a follow up.